### PR TITLE
Record new auth methods in telemetry

### PR DESCRIPTION
**Pull request by Flanker32**
Record new auth methods in telemetry

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 1]
```diff
@@ -366,24 +374,11 @@ protected boolean isAuthConfigurationExist() {
...
-            return "AzureCLI";
-        }
-        if (StringUtils.isNotEmpty(authSetting.getServerId())) {
-            return "ServerId";
-        }
-        if (authSetting.getFile() != null) {
-            return "AuthFile";
-        }
-        return "Unknown";
+        return azureTokenWrapper == null ? AUTH_TYPE_UNKNOWN : azureTokenWrapper.getAuthType().name();
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: AUTH_TYPE_UNKNOWN is confusing here

Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Fall back to old codes here to adapt users who use old authentication methods

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 2]
```diff
@@ -267,7 +274,10 @@ public Azure getAzureClient() throws AzureAuthFailureException {
             } else {
                 initAuth();
                 try {
-                    azure = AzureClientFactory.getAzureClient(getAuthTypeEnum(), isAuthConfigurationExist() ? this.auth : null, this.subscriptionId);
+                    // TODO: Enable extra Azure Environment like Azure China
+                    final AzureEnvironment environment = AzureEnvironment.AZURE;
+                    azureTokenWrapper = getAuthTypeEnum().getAzureToken(isAuthConfigurationExist() ? this.auth : null, environment);
+                    azure = AzureClientFactory.getAzureClient(azureTokenWrapper, this.subscriptionId);
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: check null azureTokenWrapper first before calling AzureClientFactory.getAzureClient

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java [hunk 3]
```diff
@@ -29,9 +26,7 @@
     public static final String SUBSCRIPTION_NOT_SPECIFIED = "Subscription ID was not specified, using the first subscription in current account," +
             " please refer https://github.com/microsoft/azure-maven-plugins/wiki/Authentication#subscription for more information";
 
-    public static Azure getAzureClient(AuthType authType, AuthConfiguration auth, String subscriptionId) throws IOException, AzureLoginFailureException {
-        final AzureEnvironment environment = AzureEnvironment.AZURE;
-        final AzureTokenWrapper azureTokenCredentials = authType.getAzureToken(auth, environment);
+    public static Azure getAzureClient(AzureTokenWrapper azureTokenCredentials, String subscriptionId) throws IOException, AzureLoginFailureException {
         if (azureTokenCredentials != null) {
```
Comment (andxu) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java]: throw NPE for null azureTokenCredentials

Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java]: Using `Preconditions.checkNotNull` to do the validation

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java (hunk 2 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Done

**Pull request merged**